posts/zzz_DO_NOT_EDIT_introducing.../introducing_sdtm.oak.qmd

```diff
@@ -0,0 +1,254 @@
+
+```{r}
+library(sdtm.oak)
+library(dplyr)
+
+cm_raw <- tibble::tribble(
+  ~oak_id, ~raw_source, ~patient_number, ~MDRAW, ~DOSU, ~MDPRIOR,
+  1L, "cm_raw", 375L, "BABY ASPIRIN", "mg", 1L,
+  2L, "cm_raw", 375L, "CORTISPORIN", "Gram", 0L,
+  3L, "cm_raw", 376L, "ASPIRIN", NA, 0L
+)
+
+study_ct <- tibble::tribble(
+  ~codelist_code, ~term_code, ~term_value, ~collected_value, ~term_preferred_term, ~term_synonyms,
+  "C71620", "C25613", "%", "%", "Percentage", "Percentage",
+  "C71620", "C28253", "mg", "mg", "Milligram", "Milligram",
+  "C71620", "C48155", "g", "g", "Gram", "Gram"
+)
+
+cm <-
+  # Derive topic variable
+  # SDTM Mapping - Map the collected value to CM. CMTRT
+  assign_no_ct(
+    raw_dat = cm_raw,
+    raw_var = "MDRAW",
+    tgt_var = "CMTRT"
+  ) %>%
+  # Derive qualifier CMDOSU
+  # SDTM Mapping - Map the collected value to CM. CMDOSU
+  assign_ct(
+    raw_dat = cm_raw,
+    raw_var = "DOSU",
+    tgt_var = "CMDOSU",
+    ct_spec = study_ct,
+    ct_clst = "C71620",
+    id_vars = oak_id_vars()
+  ) %>%
+  # Derive qualifier CMSTTPT
+  # SDTM mapping - If MDPRIOR == 1 then CM.CMSTTPT = 'SCREENING'
+  hardcode_no_ct(
+    raw_dat = condition_add(cm_raw, MDPRIOR == "1"),
+    raw_var = "MDPRIOR",
+    tgt_var = "CMSTTPT",
+    tgt_val = "SCREENING",
+    id_vars = oak_id_vars()
+  )
+```
+
+As you can see in this function call, the raw dataset and variable names are passed as arguments. As long as the raw dataset and variable are present in the global environment, the function will execute the algorithm's logic and create the target SDTM variable.
+
+{sdtm.oak} is designed to handle any type of raw input format. It is not tied to any specific data collection standards, making it both EDC-agnostic and data standards-agnostic.
+
+# Why not use {dplyr}?
+
+As you can see from the definition of the algorithms, all of them are a form of mutate statement.
+However, these functions provide a way to pass dataset and variable names as parameters and the ability to merge with the previous step by id variables.
+This enables users to build the code in a modular and simplistic fashion, mapping one SDTM variable at a time, connected by pipes.
+
+The SDTM mappings can also be used together in a single step, such as applying a filter condition, executing an mapping, and merging the outcome with the previous step.
+When there is a need to apply controlled terminology, the algorithms perform additional checks, such as verifying the presence of the value in the study's controlled terminology specification, which is passed as an object to the function call.
+If the collected value is present, it applies the standard submission value.
+
+While all these functionalities can be achieved with dplyr, {sdtm.oak} functions make it simpler to use, resulting in a modular way to build SDTM datasets.
```

Comment:
I would also add here that it is not jus simpler to use, but also more efficient and less error prone as it is working with metadata, i.e. for the controlled terminology.